scripts/verify-snapshot-source-map.js

```diff
@@ -0,0 +1,443 @@
+    }, null, 2))
+
+    const args = [configFilePath, '--log-level=warning']
+
+    await new Promise((resolve, reject) => {
+      const child = childProcess.spawn(snapbuildPath, args, { cwd: tempDir, stdio: ['pipe', 'pipe', 'inherit'] })
+      child.stdout.on('end', resolve)
+      child.on('error', reject)
+    })
+
+    let outJs
+    let outJsMap
+
+    outJs = await fs.readFile(path.join(tempDir, 'out.js'), 'utf8')
+    outJsMap = await fs.readFile(path.join(tempDir, 'out.js.map'), 'utf8')
+
+    // // Check the mapping of various key locations back to the original source
+    const checkMap = (out, map) => {
+      for (const id in toSearch) {
+        const outIndex = out.indexOf(`"${id}"`)
+        if (outIndex < 0) throw new Error(`Failed to find "${id}" in output`)
+        const outLines = out.slice(0, outIndex).split('\n')
+        const outLine = outLines.length
+        const outColumn = outLines[outLines.length - 1].length
+        const { source, line, column } = map.originalPositionFor({ line: outLine, column: outColumn })
+
+        const inSource = toSearch[id];
+        recordCheck(source === inSource, `expected source: ${inSource}, observed source: ${source}`)
+
+        const inJs = map.sourceContentFor(source)
+        let inIndex = inJs.indexOf(`"${id}"`)
+        if (inIndex < 0) inIndex = inJs.indexOf(`'${id}'`)
+        if (inIndex < 0) throw new Error(`Failed to find "${id}" in input`)
+        const inLines = inJs.slice(0, inIndex).split('\n')
+        const inLine = inLines.length
+        const inColumn = inLines[inLines.length - 1].length
+
+        const expected = JSON.stringify({ source, line: inLine, column: inColumn })
+        const observed = JSON.stringify({ source, line, column })
+        recordCheck(expected === observed, `expected original position: ${expected}, observed original position: ${observed}`)
+
+        // Also check the reverse mapping
+        const positions = map.allGeneratedPositionsFor({ source, line: inLine, column: inColumn })
+        recordCheck(positions.length > 0, `expected generated positions: 1, observed generated positions: ${positions.length}`)
+        let found = false
+        for (const { line, column } of positions) {
+          if (line === outLine && column === outColumn) {
+            found = true
+            break
+          }
+        }
+        const expectedPosition = JSON.stringify({ line: outLine, column: outColumn })
+        const observedPositions = JSON.stringify(positions)
+        recordCheck(found, `expected generated position: ${expectedPosition}, observed generated positions: ${observedPositions}`)
+      }
+    }
+
+    const sources = JSON.parse(outJsMap).sources
+    for (let source of sources) {
+      if (sources.filter(s => s === source).length > 1) {
+        throw new Error(`Duplicate source ${JSON.stringify(source)} found in source map`)
+      }
+    }
+
+    const outMap = await new SourceMapConsumer(outJsMap)
```

Comment:
This SourceMapConsumer likely needs destroyed

Reply:
Not in the JS world correct?

Reply:
Not sure what you mean?
https://www.npmjs.com/package/source-map#sourcemapconsumerprototypedestroy

Reply:
Good find! [`2d201ba` (#10)](https://github.com/cypress-io/esbuild/pull/10/commits/2d201ba451e219e2df1a90f7d1419e180cecb86b)